File: geometry-core/src/main/java/essence/geometry/core/TupleSet.java
/*
 * Copyright (C) 2018 joseluis.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
 * MA 02110-1301  USA
 */
package essence.geometry.core;

/**
 *
 * @author joseluis
 */
public interface TupleSet {

    void set(int index, byte value);

    void set(int index, short value);

    void set(int index, int value);

    void set(int index, long value);

    void set(int index, float value);

    void set(int index, double value);

    <T> void set(Class<T> type, int index, T value);
}

File: geometry-core-cs/essence/geometry/core/TupleSet.cs
namespace essence.geometry.core
{
    public interface TupleSet
    {
        void SetByte(int index, byte value);

        void SetShort(int index, short value);

        void SetInteger(int index, int value);

        void SetLong(int index, long value);

        void SetFloat(int index, float value);

        void SetDouble(int index, double value);

        void Set<T>(int index, T value);
    }
}
